streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java

```diff
@@ -188,6 +192,170 @@ public Long apply(K aggKey, V value, Long aggregate) {
+            public Long apply(final K aggKey, final Long aggOne, final Long aggTwo) {
+                return aggOne + aggTwo;
+            }
+        };
+        final KStreamSessionWindowAggregate<K, V, Long> sessionWindowAggregate
+                = new KStreamSessionWindowAggregate<>(sessionWindows,
+                                                      storeSupplier.name(),
+                                                      initializer,
+                                                      aggregator,
+                                                      sessionMerger);
+
+        return (KTable<Windowed<K>, Long>) doAggregate(
+                sessionWindowAggregate, AGGREGATE_NAME, storeSupplier);
+    }
+
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public KTable<Windowed<K>, V> reduce(final Reducer<V> reducer,
+                                         final SessionWindows sessionWindows,
+                                         final String storeName) {
+
+        Objects.requireNonNull(storeName, "storeName can't be null");
+        return reduce(reducer, sessionWindows,
+                      windowedStore(valSerde,
+                                    new InternalSessionWindow(sessionWindows.size(),
+                                                              sessionWindows.maintainMs()), true, storeName));
+    }
+
+    @Override
+    public KTable<Windowed<K>, V> reduce(final Reducer<V> reducer,
+                                         final SessionWindows sessionWindows,
+                                         final StateStoreSupplier<SessionStore> storeSupplier) {
+        Objects.requireNonNull(reducer, "reducer can't be null");
+        Objects.requireNonNull(sessionWindows, "sessionWindows can't be null");
+        Objects.requireNonNull(storeSupplier, "storeSupplier can't be null");
+
+        final Initializer<V> initializer = new Initializer<V>() {
+            @Override
+            public V apply() {
+                return null;
+            }
+        };
+
+        final Aggregator<K, V, V> aggregator = new Aggregator<K, V, V>() {
+            @Override
+            public V apply(final K aggKey, final V value, final V aggregate) {
+                if (aggregate == null) {
+                    return value;
+                }
+                return reducer.apply(aggregate, value);
+            }
+        };
+
+        final SessionMerger<K, V> sessionMerger = new SessionMerger<K, V>() {
+            @Override
+            public V apply(final K aggKey, final V aggOne, final V aggTwo) {
+                return aggregator.apply(aggKey, aggTwo, aggOne);
+            }
+        };
+
+        return aggregate(initializer, aggregator, sessionMerger, sessionWindows, valSerde, storeSupplier);
+    }
+
+
+
+    private <W extends Window, T> StateStoreSupplier windowedStore(final Serde<T> aggValSerde,
+                                                                   final Windows<W> windows,
```

Comment:
I take it back - it isn't used anywhere else. The method actually moved to `AbstractStream`, bad merge.